lib/Dialect/FIRRTL/FIRRTLUtils.cpp

```diff
@@ -823,3 +823,59 @@ Type circt::firrtl::lowerType(Type type) {
 
   return {};
 }
+
+Type circt::firrtl::lowerType(Type type, hw::HWModuleOp module) {
```

Comment:
Introducing ops in the parent scope is a very different safety criteria than creating types.  This has lots of subtle bug potential.

Reply:
Moved the addition of typedecl ops to a thread safe utility in `LowerToHW`